plugins/modules/aci_interface_policy_leaf_fc_policy_group.py

```diff
@@ -0,0 +1,361 @@
+        description=dict(type="str", aliases=["descr"]),
+        fibre_channel_interface_policy=dict(type="str", aliases=["fibre_channel_interface_policy_name"]),
+        port_channel_policy=dict(type="str", aliases=["port_channel_policy_name"]),
+        attached_entity_profile=dict(type="str", aliases=["aep_name", "aep"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["policy_group"]],
+            ["state", "present", ["policy_group"]],
+        ],
+    )
+
+    policy_group = module.params.get("policy_group")
+    description = module.params.get("description")
+    lag_type = module.params.get("lag_type")
+    fibre_channel_interface_policy = module.params.get("fibre_channel_interface_policy")
+    port_channel_policy = module.params.get("port_channel_policy")
+    attached_entity_profile = module.params.get("attached_entity_profile")
+    state = module.params.get("state")
+    name_alias = module.params.get("name_alias")
+
+    aci = ACIModule(module)
+
+    if lag_type == "port":
+        aci_class_name = "infraFcAccPortGrp"
+        dn_name = "fcaccportgrp"
+    elif lag_type == "port_channel":
+        aci_class_name = "infraFcAccBndlGrp"
+        dn_name = "fcaccbundle"
+
+    class_config_dict = dict(
+        name=policy_group,
+        descr=description,
+        nameAlias=name_alias,
+    )
+
+    child_configs = [
+        dict(
+            infraRsFcL2IfPol=dict(
+                attributes=dict(
+                    tnFcIfPolName=fibre_channel_interface_policy,
+                ),
+            ),
+        ),
+        dict(
+            infraRsFcAttEntP=dict(
+                attributes=dict(
+                    tDn="uni/infra/attentp-{0}".format(attached_entity_profile),
+                ),
+            ),
+        ),
+    ]
+
+    # Add infraRsFcLagPol binding only when port_channel_policy iß defined
+    if lag_type == "port_channel" and port_channel_policy is not None:
+        child_configs.append(
+            dict(
+                infraRsFcLagPol=dict(
+                    attributes=dict(
+                        tnLacpLagPolName=port_channel_policy,
+                    ),
+                ),
+            )
+        )
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class=aci_class_name,
+            aci_rn="infra/funcprof/{0}-{1}".format(dn_name, policy_group),
+            module_object=policy_group,
+            target_filter={"name": policy_group},
+        ),
+        child_classes=[
+            "infraRsFcL2IfPol",
+            "infraRsFcLagPol",
+            "infraRsFcAttEntP",
+        ],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
```

Comment:
should we allow possibility of removal of children? or is this not possible for the child classes?